notebooks/supply_chain/interact_with_environment.ipynb

```diff
@@ -0,0 +1,777 @@
+      "  [consumer unit, id = 25]\n",
+      "[product unit, id = 26] (sku id = 3)\n",
+      "  [consumer unit, id = 27]\n",
+      "\n",
+      "##### Retailer_001, id = 28 #####\n",
+      "[storage unit, id = 29]\n",
+      "[product unit, id = 30] (sku id = 1)\n",
+      "  [consumer unit, id = 31]\n",
+      "  [seller unit, id = 32]\n",
+      "[product unit, id = 33] (sku id = 3)\n",
+      "  [consumer unit, id = 34]\n",
+      "  [seller unit, id = 35]\n",
+      "[product unit, id = 36] (sku id = 2)\n",
+      "  [consumer unit, id = 37]\n",
+      "  [seller unit, id = 38]\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "def show_distribution_unit(unit: dict) -> None:\n",
+    "    if unit is None:\n",
+    "        return\n",
+    "    unit_id = unit[\"id\"]\n",
+    "    unit_type = unit[\"node_name\"]\n",
+    "    print(f\"[{unit_type}, id = {unit_id}]\")\n",
+    "    for vehicle in unit[\"children\"]:\n",
+    "        unit_id = vehicle[\"id\"]\n",
+    "        unit_type = vehicle[\"node_name\"]\n",
+    "        print(f\"  [{unit_type} unit, id = {unit_id}]\")\n",
+    "\n",
+    "def show_storage_unit(unit: dict) -> None:\n",
+    "    if unit is None:\n",
+    "        return\n",
+    "    unit_id = unit[\"id\"]\n",
+    "    unit_type = unit[\"node_name\"]\n",
+    "    print(f\"[{unit_type} unit, id = {unit_id}]\")\n",
+    "\n",
+    "def show_product_units(units: dict) -> None:\n",
+    "    for sku_id, unit in units.items():\n",
+    "        unit_id = unit[\"id\"]\n",
+    "        unit_type = unit[\"node_name\"]\n",
+    "        print(f\"[{unit_type} unit, id = {unit_id}] (sku id = {sku_id})\")\n",
+    "        for child in (\"consumer\", \"manufacture\", \"seller\"):\n",
+    "            if unit[child] is None:\n",
+    "                continue\n",
+    "            child_id = unit[child][\"id\"]\n",
+    "            child_type = unit[child][\"node_name\"]\n",
+    "            print(f\"  [{child_type} unit, id = {child_id}]\")\n",
+    "            \n",
+    "# Show facilities and units\n",
+    "for facility_id, facility_info in env.summary[\"node_mapping\"][\"facilities\"].items():\n",
+    "    facility_name = facility_info[\"name\"]\n",
+    "    \n",
+    "    print(f\"##### {facility_name}, id = {facility_id} #####\")\n",
+    "    show_distribution_unit(facility_info[\"units\"][\"distribution\"])\n",
+    "    show_storage_unit(facility_info[\"units\"][\"storage\"])\n",
+    "    show_product_units(facility_info[\"units\"][\"products\"])\n",
+    "    print()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get dynamic information from the environment\n",
+    "\n",
+    "Helper functions for displaying storage and manufacture status."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_storage_status(env: Env, facility_info: dict) -> List:\n",
```

Comment:
Hmmm, "I'm afraid it would let people feel like it's not easy to use..." is a good concern...I will write more comments in this cell, telling readers these codes are just to make the output of this notebook more beautiful, not necessary. Hope this could help :)